### PR TITLE
Maya: Optimize `get_related_sets`

### PR DESCRIPTION
**Pull request by BigRoy**
## Changelog Description

Optimize `get_related_sets` by opting-out earlier if no valid sets to perform less maya calls

## Additional info

This is a bit of a micro-optimization. Just happened to have this in our OpenPype codebase.

## Testing notes:

1. Maya look collecting and publishing should still work

### CODE REVIEW

**Review by tokejepsen — APPROVED** [review]
Tested successfully in Maya 2023.

File: client/ayon_core/hosts/maya/api/lib.py [hunk 1]
```diff
@@ -2168,12 +2162,20 @@ def get_related_sets(node):
     sets = [s for s in sets if s not in deformer_sets]
 
     # Ignore when the set has a specific suffix
-    sets = [s for s in sets if not any(s.endswith(x) for x in ignore_suffices)]
+    ignore_suffices = ("out_SET", "controls_SET", "_INST", "_CON")
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/maya/api/lib.py:2165]: There are 3 list comprehensions, could that be simplified to:
```python
    # Ignore when the set has a specific suffix
    ignore_suffices = ("out_SET", "controls_SET", "_INST", "_CON")
    # Default nodes to ignore
    defaults = {"defaultLightSet", "defaultObjectSet"}
    sets = [
        s
        for s in sets
        if (
            s not in deformer_sets
            and s not in defaults
            not s.endswith(ignore_suffices)
        )
]
```

Comment (iLLiCiTiT) [reply client/ayon_core/hosts/maya/api/lib.py:2165]: Or you can use tuple in `endswith`?

Comment (BigRoy) [reply client/ayon_core/hosts/maya/api/lib.py:2165]: I could combine them - sure. 

Note that `sets = [s for s in sets if not any(s.endswith(x) for x in ignore_suffices)]` can also be written as ` sets = [s for s in sets if not s.endswith(ignore_suffices)]`. `str.endswith` can actually take a `tuple`. 🤯

Comment (iLLiCiTiT) [reply client/ayon_core/hosts/maya/api/lib.py:2165]: Cool, modified the snippet.

Comment (BigRoy) [reply client/ayon_core/hosts/maya/api/lib.py:2170]: Changed the code with https://github.com/ynput/ayon-core/pull/324/commits/f3324c3863598f1946525aef02cbd8abd5444637

File: client/ayon_core/hosts/maya/api/lib.py [hunk 2]
```diff
@@ -2159,21 +2151,31 @@ def get_related_sets(node):
...
+    sets = [s for s in sets if not s.endswith(ignore_suffices)]
+    if not sets:
+        return sets
 
     # Ignore viewport filter view sets (from isolate select and
     # viewports)
+    view_sets = get_isolate_view_sets()
     sets = [s for s in sets if s not in view_sets]
-    sets = [s for s in sets if s not in defaults]
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/maya/api/lib.py:2176]: Can't tell if this is ok, @antirotor can you confirm?

Comment (BigRoy) [reply client/ayon_core/hosts/maya/api/lib.py:2176]: The `defaults` filtering is still there - just higher up added into the `exclude_sets` `set`.

Comment (iLLiCiTiT) [reply client/ayon_core/hosts/maya/api/lib.py:2166]: But https://github.com/ynput/ayon-core/pull/324/files#diff-d3957520925f7a55982d51ee662c9f5432aaa46f67cdff06858b2d3a9575598eR2167 reset the `sets` variable, or?

Comment (BigRoy) [reply client/ayon_core/hosts/maya/api/lib.py:2166]: Note the `sets` but the `view_sets`

All the logic is just filtering the `sets` to exclude whatever we don't want.
Or maybe I'm just misreading what you might be pointing out to be a bug - I just don't see it :D

**Pull request merged**